tests/backup/backup_test.go

```diff
@@ -1951,6 +1953,111 @@ var _ = Describe("{BackupRestoreOverPeriodSimultaneous}", func() {
 	})
 }*/
 
+func generate_encryption_key() string {
+	var lower = []byte("abcdefghijklmnopqrstuvwxyz")
+	var upper = []byte("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	var number = []byte("0123456789")
+	var special = []byte("~=+%^*/()[]{}/!@#$?|")
+	allChar := append(lower, upper...)
+	allChar = append(allChar, number...)
+	allChar = append(allChar, special...)
+
+	b := make([]byte, 12)
+	// select 1 upper, 1 lower, 1 number and 1 special
+	b[0] = lower[rand.Intn(len(lower))]
+	b[1] = upper[rand.Intn(len(upper))]
+	b[2] = number[rand.Intn(len(number))]
+	b[3] = special[rand.Intn(len(special))]
+	for i := 4; i < 12; i++ {
+		// randomly select 1 character from given charset
+		b[i] = allChar[rand.Intn(len(allChar))]
+	}
+
+	//shuffle character
+	rand.Shuffle(len(b), func(i, j int) {
+		b[i], b[j] = b[j], b[i]
+	})
+
+	return string(b)
+}
+
+// This test cases creates a backup location with encryption
+var _ = Describe("{BackupLocationWithEncryptionKey}", func() {
+	var contexts []*scheduler.Context
+	var appContexts []*scheduler.Context
+	var bkpNamespaces []string
+	var backup_location_name string
+	var CloudCredUID string
+	var cluster_uid string
+	var CredName string
+	var restorename string
+	var cluster_status api.ClusterInfo_StatusInfo_Status
+	providers := getProviders()
```

Comment:
move these callls inside It , so that these are call only when test is triggered